perceval/components/processor.py

```diff
@@ -173,10 +173,8 @@ def with_input(self, input_state: BasicState) -> None:
         The properties of the source will alter the input state. A perfect source always delivers the expected state as
         an input. Imperfect ones won't.
         """
-        if 'min_detected_photons' in self._parameters:
-            self._min_detected_photons_filter = self._parameters['min_detected_photons']
         if self._min_detected_photons_filter is None and self._source.is_perfect():
```

Comment:
I think it's not the correct location to check if the simulation is perfect or not. Noise can still be injected after the call to `Processor.with_input`.
In one of yesterday's comment, I suggested this default value in the special case of a perfect simulation is set just before checking that a user value is available (i.e. at the beginning of the `probs` and `samples` methods).
Could you factor this check + special case in a method?